keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1261 @@
+* Be able to remove some of the complexity with current setup and, over time, reduce
+  the amount of code contained within Kubelet.  Instead, build a library with specific
+  needs.
+* Have a faster path to desired changes, without potentially impacting the core of 
+  Kubernetes with every policy change. <br>
+  Note that current solutions have been cropping up to allow for resource management
+  outside the core Kubelet. These solutions require turning off the Kubelet functionality 
+  and overriding current Kubelet allocation. We should provide a path otherwise. Many of these solutions are custom to particular companies. Many are not open source. The community should give a process to allow this functionality in core kubelet at a granular level, same as we have in many systems, such as HPC or telco.
+* Leverage additional parts from the pod spec information which could help make allocation decisions
+without the need to query the k8s controlplane.
+* Be able to do research, with minimum toil, on new policies and resource management strategies
+
+This design will also use gRPC, which is used for many other pluggable components within Kubernetes.  
+The hope is that it will also, as a side effect, allow a path to simplify the Kubelet as it is 
+today and move the complexity into external plugins.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+#### Alpha Goals
+
+1.	Provide Initial infrastructure to support CCI Driver Plugins and k8s Deployments requiring CCI Drivers after scheduling
+2.	Support seamless k8s start ( no plugins required) through the intermediate CCI CPU management policy (functionality will be moved in beta to None and Static policies)
+3.	Support standard pod deployments not requiring CCI Drivers through CPU Manager: best-effort QoS(stretch goal will include static handling of burstable and guaranteed QoS)
+4.	CCI Driver results passed to CPU Manager (i.e Resource sets) have to support cpusets and memory affinity, .. 
+5.	Consistent on-none state handling between CCI Driver and CPU Manager (race-free)
+6.	Support proper fail mechanisms of Pods if CCI driver is not available - Error message + retry 
+7.	Continue cluster operation if CCI driver is not available
+8.  Podspecs:  Will be able to support current pod specs.  While there may be additional 
+extensibility in the future, the current pod specs will still work.
+
+
+#### Beta & Post-Beta Goals
+
+1.	Support standard pod deployments not requiring CCI Drivers through CPU & Memory Manager: all QoS types
+2.  Support hint providers for topology manager
+3.  Interoperability with Device Plugins, DRA, et cetera. 
+4.	Identify minimal in-cluster operational core (from existing CPU Manager, Memory Manager, Topology Manager)
+5.  Replace CCI policy with a component integrated in None and Static CPU Manager policies
+6.	E2E testing including other components
```

Comment:
we can pull in testing None behavior of together with CCI Drivers and a simple CCI Driver test, i will add this points

Reply:
I agree, we need node e2e tests to validate the end to end functionality of this feature delivered in the alpha phase. Please update the `Alpha Goals` section with this information.

In the goals for for Beta, we should include e2e testing of the capabilities proposed in the Beta phase as well as include the following points:

1. Gather feedback on the feature perhaps with surveys. 
2. Adding support for e2e tests in testgrid in order for the tests to run in Kubernetes upstream test infrastructure. We would need signals on the health of this feature and investigate in case there are unexpected failures or regressions.

Reply:
added.